### ISSUE TITLE
Show latest version on homepage

### ISSUE THREAD

**Issue by pascalberger** [cake-build/website#1139]
It would be nice to have latest version shown on landing page with a link to the [runners overview](https://cakebuild.net/docs/running-builds/runners/)

**Comment by chrysillala** [comment]
Hi, I would like to work on this as part of Hacktoberfest. Could you please assign and guide me? Thanks!

**Comment by pascalberger** [comment]
@chrysillala Thanks for giving this a try.

My idea is to have something like this, where the text `Version 0.38.5 is now available` should be a link to the release blog post.

![image](https://user-images.githubusercontent.com/2190718/96353037-c801d900-10c8-11eb-96c4-05bcaf9baf8f.png)

The website is a static website built using [Wyam](https://wyam.io). We don't build website scheduled (e.g. once a day), only when changes are done. 

I see two options how to get the latest version:

1. From GitHub.

    In case of a release we will first publish GitHub release before publishing a blog post, therefore it should be possible to retrieve latest version of Cake through GitHub API (https://api.github.com/repos/cake-build/cake/releases?per_page=1) as part of the Wyam pipeline. There's a [Wyam GitHub module](https://wyam.io/modules/github) available which probably could be used to call the API. 

2. From Blog Post

    Release blog posts are in a separate category: https://cakebuild.net/blog/Release-Notes/. It should be possible to get the newest and release version from it. It might also be possible to add additional metadata to the blog posts to easier get the release name. Advantage of this approach is that you already would have the link to the blog post. The downside of this approach is that it would require a blog post for every release. Stuff like [this blog post](https://cakebuild.net/blog/2020/06/cake-v0.38.3-released-in-v0.38.3) would probably break stuff.

**Comment by chrysillala** [comment]
@pascalberger Thanks for the detailed explanation. Will keep you updated.

**Comment by chrysillala** [comment]
Hi @pascalberger, I was following the `getting-started` guide and then stuck at this step `dotnet tool install Cake.Tool --version x.y.z`. 
This is the error message: `Specified version 'x.y.z' is not a valid NuGet version range.`
Could you please help me? Thanks!

**Comment by gep13** [comment]
x.y.z is shown as an indication that you need to provide a version number. The current latest release version is 0.38.5.

**Comment by pascalberger** [comment]
@chrysillala What are you trying to achieve? The getting started guide is if you want to create a new project using Cake. You'll need to replace `x.y.z` with the version number of Cake you want to use in this case.

For building the website locally you won't need the getting started guide but instead need to follow instructions in the [README](https://github.com/cake-build/website/#building).

**Comment by chrysillala** [comment]
@gep13 @pascalberger I was trying to setup on my machine for building locally, I already install .NET Core and Wyam, is it correct? What else do I need to add? Because my machine doesn't recognize `build` command.
Really appreciate your help :)

**Comment by pascalberger** [comment]
On what OS are you running?

You need to just call `build.ps1` or `build.sh` from the repo root. Dependencies like Wyam are installed automatically.

**Comment by chrysillala** [comment]
I'm on macOS, using zsh. Just run the `build.ps1` and `build.sh` but still not found.
`command not found: build.sh` `command not found: build.ps1`

**Comment by gep13** [comment]
@chrysillala I suspect that the issue that you are running into is that you are trying to run:

```
build.sh
```

Rather than:

```
./build.sh
```

As shown in this screenshot:

![image](https://user-images.githubusercontent.com/1271146/96626447-6a0f0480-1307-11eb-9433-7cbc220cb9aa.png)

When running on a Mac, you will likely run into an issue where it shows an error about too many documents.  @pascalberger may be able to provide some steps on how to reduce the amount of generated documents, and still be able to run the site.

**Comment by pascalberger** [comment]
> When running on a Mac, you will likely run into an issue where it shows an error about too many documents. @pascalberger may be able to provide some steps on how to reduce the amount of generated documents, and still be able to run the site.

I think when you just run `./build.sh --target=Preview` you should be fine with number of documents. If you run any of the `--target=Get*` targets before it will download additional artifacts (addins, source code) which will increase build time and might result in errors.

**Comment by pascalberger** [comment]
@chrysillala To make implementation simpler I would suggest to start with just showing the Cake version (retrieved through GitHub API) in the overlay, without a link to the blog post. Once this is done we can figure out how to link to the appropriate blog post.

**Comment by chrysillala** [comment]
@gep13 @pascalberger thank you for the help, but right now I'm getting another error message after running `./build.sh` or `./build.sh --target=Preview` as below
```
========================================
Build
========================================
It was not possible to find any compatible framework version
The framework 'Microsoft.NETCore.App', version '2.1.0' was not found.
  - The following frameworks were found:
      3.1.8 at [/Users/Documents/cake-website/.dotnet/shared/Microsoft.NETCore.App]

You can resolve the problem by installing the specified framework and/or SDK.

The specified framework can be found at:
  - https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=2.1.0&arch=x64&rid=osx.10.15-x64
An error occurred when executing task 'Build'.
Error: One or more errors occurred. (Wyam: Process returned an error (exit code 150).)
        Wyam: Process returned an error (exit code 150).
```

It means I need to get the 'Microsoft.NETCore.App', version '2.1.0', right? But when I click the link, they only have Windows and Linux Installer..

Apologize that I still need to go back and forth with you since I'm not familiar with the tools.
Would appreciate your help. Thanks!

**Comment by devlead** [comment]
@chrysillala have you fetched the latest commits from repo? A fix from @gep13 was submitted earlier today.

**Comment by pascalberger** [comment]
@chrysillala Are you still interested in working on this?

**Comment by augustoproiete** [comment]
@pascalberger I can take this one if @chrysillala is no longer interested.

**Comment by pascalberger** [comment]
@augustoproiete Thanks, I assigned this issue to you

**Comment by chrysillala** [comment]
Hi everyone, apologize for the late reply. I'm sorry that I couldn't continue working on the issue since I'm still having trouble setting up the repo, not sure why though I already followed your instruction. Hopefully we can work together another time :)
